src/app/trip-calendar/trip-calendar.component.ts

```diff
@@ -1,38 +1,69 @@
-    departureGate = DEPARTURE_AIRCRAFT_STATUS.gate;
-
-    returnBoardingMonth = MONTHS[RETURN_AIRCRAFT_STATUS.gateOpenTime.getMonth()];
-    returnBoardingDateTimeString = getDateAsDDTTTT(RETURN_AIRCRAFT_STATUS.gateOpenTime);
-    returnMonth = MONTHS[RETURN_AIRCRAFT_STATUS.estimatedBoardingTime.getMonth()];
-    returnDateTimeString = getDateAsDDTTTT(RETURN_AIRCRAFT_STATUS.estimatedBoardingTime);
-    returnGate = RETURN_AIRCRAFT_STATUS.gate;
+    departureBoardingMonth: any;
+    departureBoardingDateTimeString: any;
+    departureMonth: any;
+    departureDateTimeString: any;
+    departureGate: any;
+
+    returnBoardingMonth: any;
+    returnBoardingDateTimeString: any;
+    returnMonth: any;
+    returnDateTimeString: any;
+    returnGate: any;
+
     legendItems: LegendItem[] = [
         { icon: "color-fill", color: "legend-item__icon--background legend-item__icon__border legend-item__icon__border--now", text: "TODAY" },
         { icon: "color-fill", color: "legend-item__icon--selected legend-item__icon__border legend-item__icon__border--selected", text: "TRIP_DATES" }
     ];
+
+    constructor(private appService: AppService) { }
+
+    async ngOnInit() {
+        combineLatest([this.appService.getCurrentTrip(), this.appService.getReturnTrip(), this.appService.getDepartureAircraftStatus(), this.appService.getReturnAircraftStatus()])
+            .subscribe(([currentTrip, returnTrip, departureAircraftStatus, returnAircraftStatus]) => {
+                this.startDate = new Date(currentTrip.departureTime);
+                this.endDate = new Date(returnTrip.arrivalTime);
+                this.datesBetween = getDatesArray(this.startDate, this.endDate);
+
+                this.departureAircraftStatus = departureAircraftStatus;
+                this.returnAircraftStatus = returnAircraftStatus;
+
+                this.departureBoardingMonth = MONTHS[new Date(this.departureAircraftStatus.gateOpenTime).getMonth()];
+                this.departureBoardingDateTimeString = getDateAsDDTTTT(new Date(this.departureAircraftStatus.gateOpenTime));
+                this.departureMonth = MONTHS[new Date(this.departureAircraftStatus.departureTime).getMonth()];
+                this.departureDateTimeString = getDateAsDDTTTT(new Date(this.departureAircraftStatus.departureTime));
+                this.departureGate = this.departureAircraftStatus.gate;
+
+                this.returnBoardingMonth = MONTHS[new Date(this.returnAircraftStatus.gateOpenTime).getMonth()];
+                this.returnBoardingDateTimeString = getDateAsDDTTTT(new Date(this.returnAircraftStatus.gateOpenTime));
+                this.returnMonth = MONTHS[new Date(this.returnAircraftStatus.estimatedBoardingTime).getMonth()];
+                this.returnDateTimeString = getDateAsDDTTTT(new Date(this.returnAircraftStatus.estimatedBoardingTime));
+                this.returnGate = this.returnAircraftStatus.gate;
+
+                this.isDataAvailable = true;
+            });
```

Comment:
In this particular case it might not matter, because we have only one page, but in general unsubscribe from EVERY subscription, does not matter if Angular finalizes them.
Imagine this case:
- you subscribe to HTTP call
- while call is not finished, user decided to change the route or some conditions met, which Destroyed the caller component.
In that scenario, your HTTP call will continue to live, whereas if you had unsubscribed from it upon component destruction, HTTP call would have been cancelled